clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -399,8 +399,13 @@ private Optional<MetadataResponse.PartitionMetadata> updateLatestMetadata(
                 // Between the time that a topic is deleted and re-created, the client may lose track of the
                 // corresponding topicId (i.e. `oldTopicId` will be null). In this case, when we discover the new
                 // topicId, we allow the corresponding leader epoch to override the last seen value.
-                log.info("Resetting the last seen epoch of partition {} to {} since the associated topicId changed from {} to {}",
-                         tp, newEpoch, oldTopicId, topicId);
+                if (oldTopicId != null) {
+                    log.info("Resetting the last seen epoch of partition {} to {} since the associated topicId changed from {} to {}",
+                            tp, newEpoch, oldTopicId, topicId);
+                } else {
+                    log.debug("Resetting the last seen epoch of partition {} to {} since the associated topicId was undefined but is now set to {}",
```

Comment:
The change should remove the logging on startup. Which is where the main noise was found (ie, null to having a topic ID)

Are you still seeing that log message? If so which version are you running on?

Reply:
Thanks for the response @jolshan. I'm seeing this log statement on multiple Kafka producers. It is logged once every 5 minutes, for each topic partition, not just on startup. (Came to my attention because we exhausted our log quota and this was the biggest cause of logs).

Here is an example log that is generated:
```
[Producer clientId=producer-1] Resetting the last seen epoch of partition my.topic-3 to 1286 since the associated topicId changed from null to lkaiBwHxQ6iQHE49_AVysQ
```

This is generated every 5 minutes for each topic partition (so a topic with 10 partitions generates 10 log entries). Each time the topic id is updated from `null` to `lkaiBwHxQ6iQHE49_AVysQ`. All partitions for the same topic show the same topic id.

Running kafka-client 3.4.0 or 3.5.0 on the producers.  Brokers are running Kafka 2.8.1.

Reply:
Hmmm. This would not be in the 3.4 release right? Let's make sure it's not 3.5

Reply:
As for seeing it every 5 minutes -- that's a bit strange. Once the client receives the topic ID in the cache, it should stay. Are you creating new topics, restarting the clients, or moving partitions?

Reply:
nope. Not creating new topics, restarting or moving partitions. Looking at our organisation logs across all services, it appears to affect a lot of producers across different teams.

Reply:
That is quite strange. I have not heard of this being an issue outside of starting producers for the first time. Please ensure you are running on 3.5 and let me know if it's still an issue.